freeipa/src/main/java/com/sequenceiq/freeipa/util/CrnService.java

```diff
@@ -69,4 +70,12 @@ public String createCrn(String accountId, Crn.ResourceType resourceType) {
                 .setResource(UUID.randomUUID().toString())
                 .build().toString();
     }
+
+    public static String checkUserCrn(String userCrn) {
+        if (userCrn == null) {
+            throw new BadRequestException("User CRN must be provided");
```

Comment:
BadRequestException doesn't really make sense here. Other CrnService methods throw CrnParseExceptions. This method should throw a CrnParseException to match the behavior of the other methods.
UserV1Controller should catch CrnParseException and convert those to BadRequestExceptions otherwise those exceptions will go out at 500 Internal Server Errors.

Reply:
Make sense. I can change that.